pkg/registry/core/shoot/storage/admin_kubeconfig.go

```diff
@@ -267,3 +235,79 @@ func findNewestCACertificate(results []*gardencorev1alpha1.GardenerResourceData)
 
 	return result, nil
 }
+
+func getClusterCABundle(resourceDataList gardencorev1alpha1helper.GardenerResourceDataList) ([]byte, error) {
+	caCerts := resourceDataList.Select(caCertificateSelector.Add(nameCAClusterReq))
+	if len(caCerts) == 0 {
+		// fall back to constant CA name
+		// TODO(rfranzke): Delete this in a future release.
+		ca := resourceDataList.Get(v1beta1constants.SecretNameCACluster)
+		if ca == nil {
+			return nil, fmt.Errorf("cluster certificate authority not yet provisioned")
+		}
+		caCerts = append(caCerts, ca)
+	}
+
+	var caBundle []byte
+
+	for _, data := range caCerts {
+		cert, _, err := getCADataRaw(data)
+		if err != nil {
+			return nil, fmt.Errorf("could not fetch raw CA data for %q", data.Name)
+		}
+		caBundle = append(caBundle, cert...)
+	}
+
+	if len(caBundle) == 0 {
+		return nil, fmt.Errorf("cluster certificate authority not yet provisioned")
+	}
+
+	return caBundle, nil
```

Comment:
in the end `caBundle` should equal to `ca.crt` of secret `<shootname>.ca-cluster`, right?

Reply:
Yes, right 👍🏻